third_party/amd/lib/TritonAMDGPUToLLVM/TargetInfo.cpp

```diff
@@ -103,34 +105,183 @@ Value TargetInfo::loadDShared(RewriterBase &rewriter, Location loc, Value ptr,
 
 Value TargetInfo::shuffleXor(RewriterBase &rewriter, Location loc, Value val,
                              int i) const {
-  return LLVM::AMD::shuffleXor(loc, rewriter, val, i);
+  return LLVM::AMD::shuffleXor(loc, rewriter, val, i, getISAFamily());
 }
 
 Value TargetInfo::shuffleUp(RewriterBase &rewriter, Location loc, Value val,
                             int i) const {
-  return LLVM::AMD::shuffleUp(loc, rewriter, val, i);
+  return LLVM::AMD::shuffleUp(loc, rewriter, val, i, getISAFamily());
 }
 
 Value TargetInfo::shuffleIdx(RewriterBase &rewriter, Location loc, Value val,
                              int i) const {
-  return LLVM::AMD::shuffleIdx(loc, rewriter, val, i);
+  return LLVM::AMD::shuffleIdx(loc, rewriter, val, i, getISAFamily());
 }
 
 Value TargetInfo::shuffleIdx(RewriterBase &rewriter, Location loc, Value val,
                              Value i) const {
-  return LLVM::AMD::shuffleIdx(loc, rewriter, val, i);
+  return LLVM::AMD::shuffleIdx(loc, rewriter, val, i, getISAFamily());
 }
 
 Value TargetInfo::programId(RewriterBase &rewriter, Location loc,
                             ModuleOp moduleOp, int axis) const {
   return LLVM::AMD::llGetPid(loc, rewriter, moduleOp, axis);
 }
 
+static inline Type castToInt(RewriterBase &rewriter, Location loc, Value &val,
```

Comment:
When adding new functions, please add their documentation to describle what this function is doing with its parameters and returning what (if the function is not very trivial). Also typically it's better to make the function name and function parameter more descriptive. Right now `castToInt` is missing part of what this function is doing--`castToAndSExtInt` is more descriptive. For `bits`, naming it as `toBit` would be more clear. 

This might be small things but it helps code readability and maintenance.

Reply:
Regarding the function documentation, is there any good example I can follow? There're many different styles I'm not sure which one I should use.

Reply:
Follow upstream mlir codebase, e.g., https://github.com/llvm/llvm-project/blob/main/mlir/include/mlir/IR/OpDefinition.h.............